### ISSUE TITLE
PLW1514: extend or change autofix with `encoding="utf-8"`

### ISSUE THREAD

**Issue by xmo-odoo** [astral-sh/ruff#12069]
#8883 requested that the `open`/`read` calls without an explicit encoding be autofixed to use the locale, however *the use of the locale is exactly why PEP 597 / EncodingWarning was implemented*: in this day and age, and especially when interacting with internal data or software-created files, following the possibly wonky and misconfigured locale of the host machine is actively detrimental. Even more so when accounting for Windows where the locale is essentially always useless (it generally defaults to a single-byte windows codepage, changing the locale to UTF8 -- codepage 65001 -- is possible but uncommon).

At https://docs.astral.sh/ruff/rules/unspecified-encoding/ it is claimed that this is the PEP 597 recommendation, but it is the opposite of reality:

> [How to Teach This](https://peps.python.org/pep-0597/#how-to-teach-this)
>
> [For new users](https://peps.python.org/pep-0597/#for-new-users)
>
> Since EncodingWarning is used to write cross-platform code, there is no need to teach it to new users.
>
> We can just recommend using UTF-8 for text files and using encoding="utf-8" when opening them.

Using `encoding=locale.getpreferredencoding(False)` is a *forward-compatibility* measure, and the new `encoding="locale"` is a shorter and less error prone to specify that, but *using the locale is not the expected default*.

As a result, the autofixer for PLW1514 should either:

- autofix to `encoding-"utf-8"` by default, that is more likely to be correct, expected, and desired than not
- if supported, provide a second(ary) autofix to use the locale instead

**Comment by Avasam** [comment]
I thought I raised something about this already somewhere, but I can't find it anymore.
Even though I originally suggested the autofix for this rule (https://github.com/astral-sh/ruff/issues/8883) based on official doc. After trying it out in setuptools, I noticed that this autofix doesn't help, because `locale.getpreferredencoding(False)` will itself lead to a warning !

A better autofix imo is `encoding="locale" if sys.version_info >= (3, 10) else None` which does successfully keep the same behaviour, removes the runtime warnings *and* is easy to search to autofix to "utf-8" in a manual pass.

Alternatively, the prefered encoding could be configurable, so that "locale", "utf-8", or a project's custom needs is used for autofixing.
